### PR TITLE
Make address offset fixup configurable

### PR DESCRIPTION
**Pull request by hanno-becker**

### CODE REVIEW

File: slothy/core/config.py [hunk 1]
```diff
@@ -1111,6 +1141,11 @@ def __init__(self, Arch, Target):
         self._selfcheck_failure_logfile = None
         self._allow_useless_instructions = False
 
+        # TODO: This should be False by default, but this is a breaking
+        # change that requires a lot of examples (where it _is_ safe to
+        # apply address offset fixup) to be changed.
+        self._unsafe_address_offset_fixup = True
```
Comment (mkannwischer) [inline slothy/core/config.py:1147]: Do we want to track this in an issue or is this going to be  around forever?

File: slothy/targets/aarch64/aarch64_neon.py
Comment (mkannwischer) [inline slothy/targets/aarch64/aarch64_neon.py:1]: the same changes should also be applied to arch_v81m.py I suppose?

**Review by mkannwischer — CHANGES_REQUESTED** [review]
Can you please also add the `input` -> `in_out` changes to `arch_v81m.py`, please?

What's the reasonf or not updating the following functions in aarch64_neon.py?
- q_ldr_with_inc_hint
- q_str_with_inc_hint
- x_ldp_with_inc_hint
- x_ldp_sp_with_inc_hint
- x_ldp_sp_with_inc_hint2
- x_ldp_with_inc_hint2
- x_stp_with_inc_hint
- x_stp_sp_with_inc_hint
- x_stp_sp_with_inc_hint2
- x_stp_with_inc_hint2

File: slothy/core/config.py (hunk 1 above)
Comment (hanno-becker) [reply slothy/core/config.py:1147]: Can you open an issue?

Comment (mkannwischer) [reply slothy/core/config.py:1147]: done : https://github.com/slothy-optimizer/slothy/issues/101

**Comment by hanno-becker** [comment]
> What's the reasonf or not updating the following functions in aarch64_neon.py?
> q_ldr_with_inc_hint
> q_str_with_inc_hint
> x_ldp_with_inc_hint
> x_ldp_sp_with_inc_hint
> x_ldp_sp_with_inc_hint2
> x_ldp_with_inc_hint2
> x_stp_with_inc_hint
> x_stp_sp_with_inc_hint
> x_stp_sp_with_inc_hint2
> x_stp_with_inc_hint2

@mkannwischer Those do not alter the address register.

File: slothy/targets/aarch64/aarch64_neon.py
Comment (hanno-becker) [reply slothy/targets/aarch64/aarch64_neon.py:1]: In principle, yes, but in practice, I don't see an easy way to adjust arch_v81m.py due to the adhoc parsing and writing.

**Comment by hanno-becker** [comment]
@mkannwischer Adjusting arch_v81m.py is out of scope I'm afraid. We will need to document and check this limitation for Armv8.1-M for now, and open a separate issue.

File: slothy/targets/aarch64/aarch64_neon.py
Comment (mkannwischer) [reply slothy/targets/aarch64/aarch64_neon.py:1]: okay, let's open an issue for that then.

**Review by mkannwischer — APPROVED** [review]

File: slothy/targets/aarch64/aarch64_neon.py
Comment (hanno-becker) [reply slothy/targets/aarch64/aarch64_neon.py:1]: https://github.com/slothy-optimizer/slothy/issues/106

**Pull request merged**